torch/nn/modules/module.py

```diff
@@ -2005,6 +2022,11 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
             exists in :attr:`state_dict`, :meth:`load_state_dict` will raise a
             ``RuntimeError``.
         """
+        if assign:
+            warnings.warn("If using ``load_state_dict`` with ``assign=True``, "
```

Comment:
From offline discussion, I think we should move this to Optimizer itself when a call to step() would do no actual work.

Reply:
Actually, looking more into how the optimizer is written, 

what happens is that for each param group, an optimizer will call `[init_group](https://github.com/pytorch/pytorch/blob/main/torch/optim/sgd.py#L73)` which updates the list of `params_with_grad` and then calls the optimizer step on that param group.

I wonder whether it might be a common use case of param_groups to freeze one param group and train another (i.e. some params_with_grad lists will be intentionally empty). If this is indeed a common use case and we check whether this list is empty and warn then, it might be somewhat noisy.

We could alternatively keep a separate list of all the params across all param groups and warn at the end if this list is empty, but I feel like this is somewhat wasteful.

Are we happy with maybe just warning in the documentation?